src/collisions_updater.cpp

```diff
@@ -0,0 +1,228 @@
+    buffer.reserve(stream.tellg());
+    stream.seekg(0, std::ios::beg);
+    buffer.assign( (std::istreambuf_iterator<char>(stream)), std::istreambuf_iterator<char>() );
+    stream.close();
+  }
+  return true;
+}
+
+class SortableDisabledCollision {
+     const srdf::Model::DisabledCollision dc_;
+     const std::string key_;
+public:
+    SortableDisabledCollision(const srdf::Model::DisabledCollision &dc)
+    : dc_(dc), key_(dc.link1_ < dc.link2_ ? (dc.link1_ + "|" + dc.link2_) : (dc.link2_ + "|" + dc.link1_)) {}
+    operator const srdf::Model::DisabledCollision () const { return dc_; }
+    bool operator < (const SortableDisabledCollision &other) const { return key_ < other.key_; }
+};
+
+class CollisionUpdater{
+    moveit_setup_assistant::MoveItConfigData config_data;
+public:
+    bool loadSetupAssistantConfig(const std::string &path){
+        config_data.config_pkg_path_ = path;
+
+        fs::path setup_assistant_file = config_data.config_pkg_path_;
+        setup_assistant_file /= ".setup_assistant";
+
+        if(!config_data.inputSetupAssistantYAML( setup_assistant_file.make_preferred().native().c_str() ) )
+            return false;
+
+        fs::path urdf_path;
+
+        if( config_data.urdf_pkg_name_.empty() || config_data.urdf_pkg_name_ == "\"\"" ){
+            urdf_path = config_data.urdf_pkg_relative_path_;
+        }else{
+            fs::path robot_desc_pkg_path = ros::package::getPath( config_data.urdf_pkg_name_ );
+            if( robot_desc_pkg_path.empty() ) return false;
+
+            urdf_path = robot_desc_pkg_path / config_data.urdf_pkg_relative_path_;
+        }
+        config_data.urdf_path_ = urdf_path.make_preferred().native();
+
+        fs::path srdf_path = config_data.config_pkg_path_ ;
+        srdf_path /= config_data.srdf_pkg_relative_path_;
+        config_data.srdf_path_ = srdf_path.make_preferred().native();
+
+        return true;
+    }
+
+    void setURDF(const std::string &path) {
+        if(!path.empty()) config_data.urdf_path_ = path;
+    }
+    void setSRDF(const std::string &path) {
+        if(!path.empty()) config_data.srdf_path_ = path;
+    }
+
+    bool setup(bool keep_old, const std::vector<std::string> &xacro_args){
+        std::string urdf_string;
+        if(config_data.urdf_path_.empty() || !loadFileToString(urdf_string, config_data.urdf_path_, xacro_args)) return false;
+        if(!config_data.urdf_model_->initString( urdf_string))return false;
+
+        std::string srdf_string;
+        if(config_data.srdf_path_.empty() || !loadFileToString(srdf_string, config_data.srdf_path_, xacro_args)) return false;
+        if(!config_data.srdf_->initString( *config_data.urdf_model_, srdf_string)) return false;
+
+
+        if(!keep_old) config_data.srdf_->disabled_collisions_.clear();
+
+        return true;
+    }
+
+    moveit_setup_assistant::LinkPairMap compute(uint32_t trials, double min_collision_fraction,  bool verbose){
+        unsigned int collision_progress;
```

Comment:
explain why this variable is not used (because you have no gui to display progress)